Add postman import spec

ll931217 opened the pull request:
This PR focuses on having the Postman Collection import functionality. The following is how one can use this functionality:

Flags
Flag          | Values               | Description
------------- | -------------------- | -------------------------------------------
`-t/--type`   | `openapi`, `postman` | Pick which API spec to import
`-o/--output` |                      | Select the directory to output the generated posting requests and variables. If this isn't supplied then the directories and request files will be created where the `spec_file` is located.

Usage:
```bash
posting import -t postman postman.collection.json
```

```bash
posting import -t postman -o requests postman.collection.json
```

I wrote a test for it but I am not sure how to include it in the test workflow, the test passes when I run 

```bash
uv run pytest -d tests/test_postman_import.py
```

![image](https://github.com/user-attachments/assets/f474b2ca-ebb1-4388-a943-8898456cf0e3)


NOTE: I have tested it with [Postman API](https://www.postman.com/postman/workspace/published-postman-templates/collection/9065401-ff29b3be-af69-4442-91e0-c1158b620fc2) Collection and my company's postman collection, seems to all work.

darrenburns commented:
Thanks for this very important feature! I'm on a break at the moment but will try to take this for a spin within the next couple of weeks.

darrenburns commented on src/posting/__main__.py:
```diff
@@ -100,14 +101,29 @@ def locate(thing_to_locate: str) -> None:
+            collection = import_openapi_spec(spec_path)
+        elif type.lower() == "postman":
+            spec_type = "Postman"
+            console.print(
+                "Importing Postman collection haven't been implemented yet", style="red"
+            )
```
```suggestion
```

darrenburns commented on src/posting/importing/postman.py:
```diff
@@ -0,0 +1,180 @@
+    return env_file
+
+
+def generate_directory_structure(
+    items: List[RequestItem], current_path: str = "", base_path: Path = Path("")
+) -> List[str]:
+    directories = []
+    for item in items:
+        if item.item is not None:
+            folder_name = item.name
+            new_path = f"{current_path}/{folder_name}" if current_path else folder_name
+            full_path = Path(base_path) / new_path
+            os.makedirs(str(full_path), exist_ok=True)
+            directories.append(str(full_path))
+            generate_directory_structure(item.item, new_path, base_path)
+        if item.request is not None:
+            request_name = re.sub(r"[^A-Za-z0-9\.]+", "", item.name)
+            file_name = f"{request_name}.posting.yaml"
+            full_path = Path(base_path) / current_path / file_name
+            create_request_file(full_path, item)
+    return directories
```
Ideally we'd build a `Collection` just like the "import from OpenAPI" feature, then just call `save_to_disk`.

That would also tidy up the `import_spec` function, as it would mean that handling would be consistent across OpenAPI and Postman.

ll931217 replied:
Ah, I see! Let me update that

darrenburns commented:
Pushed a fix for a crash I got when I downloaded a sample Postman collection - `PostmanCollection.info` and `PostmanCollection.variable` were optional.

ll931217 commented:
> Pushed a fix for a crash I got when I downloaded a sample Postman collection - `PostmanCollection.info` and `PostmanCollection.variable` were optional.
> 
> Also from testing it seems like the directory structure is not being respected?

Sorry been quite busy lately, will work on this when I get time

github-carlos commented:
Hello, guys. Any predictions for closing this pull request? This is a very good feature for switching easily to posting :D

ll931217 commented:
> Hello, guys. Any predictions for closing this pull request? This is a very good feature for switching easily to posting :D

Ah, sorry about the delay, was working 2 jobs and recently my main job wasn't doing well financially so I resigned and busy doing interview preparations again as well as doing my side job. Will probably be able to fix this up when I have free time, probably in a month.

It is only missing a test, so if someone can help add it in, would be appreciated.

ll931217 commented:
@darrenburns Hey Darren, I fixed up the test and also fixed up the import spec function some more with the latest changes. Let me know what you think, thanks and sorry for the delay

darrenburns commented:
Thanks for this. I'll try to review it tomorrow.

darrenburns commented:
Sorry for the delay - I fixed up a few issues I found in this PR (the collection was being saved twice in 2 different locations, as was the README). Should be able to merge and release this soon (likely this week).
